pkg/appgateserver/config/appgate_configs_reader_test.go

```diff
@@ -0,0 +1,115 @@
+package config_test
+
+import (
+	"net/url"
+	"testing"
+
+	sdkerrors "cosmossdk.io/errors"
+	"github.com/gogo/status"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/appgateserver/config"
+	"github.com/pokt-network/poktroll/testutil/yaml"
+)
+
+func Test_ParseAppGateConfigs(t *testing.T) {
+	tests := []struct {
+		desc     string
+		err      *sdkerrors.Error
+		expected *config.AppGateConfig
+		config   string
+	}{
+		// Valid Configs
+		{
+			desc: "appgate_config_test: valid app gate config",
+			err:  nil,
+			expected: &config.AppGateConfig{
+				SelfSigning:       true,
+				SigningKey:        "app1",
+				ListeningEndpoint: &url.URL{Scheme: "http", Host: "localhost:42069"},
+				QueryNodeUrl:      &url.URL{Scheme: "tcp", Host: "127.0.0.1:36657"},
+			},
+			config: `
+				self_signing: true
+				signing_key: app1
+				listening_endpoint: http://localhost:42069
+				query_node_url: tcp://127.0.0.1:36657
+				`,
+		},
+		{
+			desc: "appgate_config_test: valid app gate config with undefined self signing",
```

Comment:
If we're adding the same prefix everywhere, no need to add it. It'll show up w/ the test name.
```suggestion
			desc: "valid app gate config with undefined self signing",
```